test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlReaderTests.cs

```diff
@@ -1699,5 +1699,39 @@ public void ImportSchemaElementsOnlyFromReferencedNamespacesl()
             var personType = model.FindType("Other.Types.Person");
             Assert.Null(personType);
         }
+
+        [Fact]
+        public void ReadKeyAlias()
+        {
+            var csdl = @"<?xml version=""1.0"" encoding=""utf-8""?>
+            <edmx:Edmx Version=""4.0"" xmlns:edmx=""http://docs.oasis-open.org/odata/ns/edmx"">
+              <edmx:DataServices>
+                <Schema Namespace=""NS1"" xmlns=""http://docs.oasis-open.org/odata/ns/edm"">
+                  <EntityType Name=""Product"">
+                    <Key>
+                      <PropertyRef Name=""ExampleObject/Name"" Alias=""ExampleObjectName""/>
+                    </Key>
+                    <Property Name=""ExampleObject"" Type=""NS1.ExampleComplexObject"" Nullable=""false""/>
+                    <Property Name=""Name"" Type=""Edm.String"" Nullable=""false""/>
+                    <Property Name=""UpdatedTime"" Type=""Edm.Date"" Nullable=""false""/>
+                  </EntityType>
+                  <ComplexType Name =""ExampleComplexObject"">
+                    <Property Name=""Name"" Type=""Edm.String"" Nullable=""false""/>
+                  </ComplexType>
+                  <EntityContainer Name=""Container"">
+                    <EntitySet Name=""Products"" EntityType=""NS1.Product""/>
+                  </EntityContainer>
+                </Schema>
+              </edmx:DataServices>
+            </edmx:Edmx>
+            ";
+
+            var model = CsdlReader.Parse(XElement.Parse(csdl).CreateReader());

```

Comment:
`CreateReader` returns an object that implements `IDisposable`. Please wrap it in a `using` block.